### ISSUE TITLE
pruning and piggyback mask training order

### ISSUE THREAD

**Issue by HaoranZhuExplorer** [ivclab/CPG#4]
Hi, I'm reading the source code and your paper. 
In your paper, it seems that for a new task, you firstly train (a piggyback mask + released space from gradual pruning for previous tasks), then you apply gradual pruning for the new task.

However, in your code(https://github.com/ivclab/CPG/blob/cpg_face/CPG_cifar100_main_normal.py), it seems that you are firstly doing fine-tuning, then gradual pruning, finally piggyback retraining. 

Why the order is different?

**Comment by HaoranZhuExplorer** [comment]
The author answered my question by email. Basically, piggyback has already been included in the finetune step, notice the parameter lr_mask. And the author told me in the email that the following piggyback retraining step is made to see if we can improve the performance more when we retraining the piggyback mask.
